### PR TITLE
Align dependencies

### PR DESCRIPTION
**Pull request by sestegra**
Update following dependencies

```
googleapis: '>=0.50.0 <0.51.0'
googleapis_beta: '>=0.45.0 <0.46.0'
googleapis_auth: '>=0.2.3 <0.3.0'
test: '>=0.12.0 <0.13.0'
```

Some e2e tests don't pass, but it's the same as previous version.

### CODE REVIEW

**Comment by sestegra** [comment]
See below test errors

```
00:07 +30 -1: test/storage/e2e_test.dart: bucket create-with-predefined-acl-delete [E]
  Expected: <0>
    Actual: <2>
```

```
0:31 +83 -2: test/db_all_e2e_test.dart: datastore_test e2e_datastore query query [E]
  ApplicationError: no matching index found. recommended index is:
  - kind: TestQueryKind
    properties:
    - name: listproperty
    - name: test_property
      direction: desc
```

```
00:43 +103 -3: test/pubsub/pubsub_e2e_test.dart: topic create-lookup-delete [E]
  Expected: <null>
    Actual: 'dart-test-197612'
```

```
00:45 +107 -4: test/db_all_e2e_test.dart: datastore_test e2e_db query [E]
  ApplicationError: no matching index found. recommended index is:
  - kind: User
    properties:
    - name: name
      direction: desc
    - name: nickname
```

```
00:57 +109 -5: test/pubsub/pubsub_e2e_test.dart: subscription create-lookup-delete [E]
  Expected: <null>
    Actual: 'dart-test-197612'
```

```
01:49 +150 -6: test/pubsub/pubsub_e2e_test.dart: (tearDownAll) [E]
  Test terminated with leftover topics and/or subscriptions
```

**Comment by sestegra** [comment]
All tests fixed

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ pubspec.lock
 packages
 .pub
 .packages
+.idea
```
Comment (kevmoo) [inline .gitignore]: Please remove this. Should be in your `.gitignore_global` file!

File: pubspec.yaml [hunk 2]
```diff
@@ -6,14 +6,14 @@ homepage: https://github.com/dart-lang/gcloud
...
-  googleapis_beta: '>=0.10.0 <0.40.0'
+  googleapis: '>=0.50.0 <0.51.0'
+  googleapis_beta: '>=0.45.0 <0.46.0'
   http: '>=0.11.0 <0.12.0'
 dev_dependencies:
   googleapis_auth: '>=0.2.3 <0.3.0'
   http_parser: '>=2.0.0 <4.0.0'
   mime: '>=0.9.0+3 <0.10.0'
-  unittest: '>=0.11.0 <0.12.0'
+  test: '>=0.12.0 <0.13.0'
```
Comment (kevmoo) [inline pubspec.yaml]: Thanks for doing this!

File: test/common_e2e.dart [hunk 3]
```diff
@@ -92,30 +91,7 @@ Future withAuthClient(List<String> scopes, AuthCallback callback,
         .clientViaServiceAccount(creds, scopes)
         .then((http.Client client) {
       if (trace) client = new TraceClient(client);
-      return callback(project, client).whenComplete(() => client.close());
+      return callback(project, client);
```
Comment (jakobr-google) [inline test/common_e2e.dart]: Doesn't have to be now, but we should refactor this method and call it something other than `withAuthClient`, now that it doesn't manage the life-cycle of the client anymore.

File: .gitignore (hunk 1 above)
Comment (sestegra) [reply .gitignore]: Done

[the remark this answers is not in the document]
Comment (sestegra) [reply .gitignore]: Done

File: lib/src/pubsub_impl.dart [hunk 4]
```diff
@@ -418,10 +418,12 @@ class _TopicPageImpl implements Page<Topic> {
   final List<Topic> items;
 
   _TopicPageImpl(this._api, this._pageSize, pubsub.ListTopicsResponse response)
-      : items = new List(response.topics.length),
+      : items = new List(response.topics != null ? response.topics.length : 0),
```
[the remark this answers is not in the document]
Comment (sestegra) [reply lib/src/pubsub_impl.dart]: I know, but it's used nowhere in this package. So I kept the same code styling as the package.

[the remark this answers is not in the document]
Comment (jakobr-google) [reply lib/src/pubsub_impl.dart]: Perfectly fine.

File: test/common_e2e.dart (hunk 3 above)
Comment (sestegra) [reply test/common_e2e.dart]: I had to refactor the code first to remove legacy dependencies to SimpleConfiguration.
Sorry, I won't have time to do it now.

Comment (jakobr-google) [reply test/common_e2e.dart]: Don't worry, this was just as much a note for me to get back to this later. You've done plenty, thanks!

**Pull request merged**